README.md

```diff
@@ -1,3 +1,70 @@
-# csp billing adapter amazon plugin
+# CSP Billing Adapter Amazon Plugin
 
-This is a plugin for [csp-billing-adapter](https://github.com/SUSE-Enceladus/csp-billing-adapter) that provides hook implementations for Amazon csp.
+This is a plugin for
+[csp-billing-adapter](https://github.com/SUSE-Enceladus/csp-billing-adapter)
+that provides CSP hook implementations. This includes the hooks defined in the
+[csp_hookspecs.py module](https://github.com/SUSE-Enceladus/csp-billing-adapter/blob/main/csp_billing_adapter/csp_hookspecs.py).
+To enable metered billing in Amazon the container or instance is required to
+have MeterUsage permissions. This can be added with a role policy such as:
+
+```
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Sid": "CSPBillingAdapterPermissions",
+            "Effect": "Allow",
+            "Action": [
+                "aws-marketplace:MeterUsage"
+            ],
+            "Resource": "*"
+        }
+    ]
+}
+```
+
+## Meter billing
+
+The `meter_billing` function accepts a dictionary mapping of dimension name
+to usage quantity. This information is used to bill the customer for
+the product ID that is configured in the adapter. If there is an exception
+with metered billing the exception is raised.
+
+## Get CSP Name
+
+The `get_csp_name` returns the name of the CSP provider. In this case it
```

Comment:
The other two descriptions meter_billing and get_account_info  have the word "function" after the quoted name. Should this have it also to be consistent?